### PR TITLE
LLM: update llm benchmark scripts.

### PR DESCRIPTION
**Pull request by lalalapotter**
## Description

This PR is to update llm benchmark scripts.
Change log:
1. add `use_cache=True` in generate method.
2. add transformer bf16 implementation.
3. add Llama2-70b temp workaround for issue: https://github.com/analytics-zoo/nano/issues/563, disable `optimize_model`
4. change tokenizer for Llama family models, AutoTokenizer --> LlamaTokenizer.

### CODE REVIEW

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 1]
```diff
@@ -45,6 +45,8 @@ def run_model(repo_id, test_api, in_out_pairs, local_model_hub=None, warm_up=1,
         result = run_transformer_int4_gpu(repo_id, local_model_hub, in_out_pairs, warm_up, num_trials)
     elif test_api == 'optimize_model_gpu':
         result = run_optimize_model_gpu(repo_id, local_model_hub, in_out_pairs, warm_up, num_trials)
+    elif test_api == 'transformer_bf16':
```
Comment (jason-dai) [inline python/llm/dev/benchmark/all-in-one/run.py:48]: change to pytorch_autocast_bf16

Comment (lalalapotter) [reply python/llm/dev/benchmark/all-in-one/run.py]: fixed.

**Comment by jason-dai** [comment]
We should probably use autocast (bf16) in our transformer_int4 benchmarking too.

**Comment by jenniew** [comment]
Can you also update config.yaml to add corresponding test_api "pytorch_autocast_bf16"

**Comment by lalalapotter** [comment]
After test transformer int4 with `torch.autocast()`, we find the latency performance of most models are worse than without `torch.autocast()`. Therefore we merge the PR without adding `torch.autocast()` for now, and will investigate it [here](https://github.com/analytics-zoo/nano/issues/578) later.

**Review by hkvision — APPROVED** [review]

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 2]
```diff
@@ -115,6 +117,17 @@ def run_transformer_int4(repo_id,
     if repo_id in ['THUDM/chatglm-6b', 'THUDM/chatglm2-6b']:
         model = AutoModel.from_pretrained(model_path, load_in_4bit=True, trust_remote_code=True, torch_dtype='auto')
         tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)
+    elif repo_id in ['meta-llama/Llama-2-70b-chat-hf']:
+        # Can be remove when issue https://github.com/analytics-zoo/nano/issues/563 is resolved.
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:121]: remove -> removed

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 3]
```diff
@@ -115,6 +117,17 @@ def run_transformer_int4(repo_id,
...
+        model = AutoModelForCausalLM.from_pretrained(model_path, load_in_4bit=True,
+                                                     trust_remote_code=True, optimize_model=False)
+        tokenizer = LlamaTokenizer.from_pretrained(model_path, trust_remote_code=True)
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:124]: Add a comment why we must use LlamaTokenizer?

**Pull request merged**